src/main/java/com/epam/izh/rd/online/repository/SimpleSchoolBookRepository.java

```diff
@@ -0,0 +1,53 @@
+package com.epam.izh.rd.online.repository;
+
+
+import com.epam.izh.rd.online.entity.Book;
+import com.epam.izh.rd.online.entity.SchoolBook;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class SimpleSchoolBookRepository implements BookRepository<SchoolBook> {
+    SchoolBook[] books = new SchoolBook[0];
+
+    @Override
+    public boolean save(SchoolBook book) {
+        SchoolBook[] newBooks = new SchoolBook[books.length + 1];
+        System.arraycopy(books, 0, newBooks, 0, books.length);
+        newBooks[books.length] = book;
+        books = newBooks;
+        return true;
+    }
+
+    @Override
+    public SchoolBook[] findByName(String name) {
+        SchoolBook[] foundBooks;
+        foundBooks = Arrays.stream(books)
+                .filter(book -> book.getName().equals(name))
+                .toArray(SchoolBook[]::new);
+        return foundBooks;
```

Comment:
nice solution with lambda! you also could inline this code and avoid useless variable 'foundBooks'